File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const N = +input[0];
const M = +input[2];
const cards = input[1].split(' ').map(Number);
const cases = input[3].split(' ').map(Number);
const countMap = new Map();

for (let card of cards) {
    if (countMap.has(card)) {
        countMap.set(card, countMap.get(card) + 1);
    } else {
        countMap.set(card, 1);
    }
}

let answer = [];

for (let c of cases) {
    answer.push(countMap.has(c) ? countMap.get(c) : 0);
}

console.log(answer.join(' '));

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
import sys

input = sys.stdin.readlines();
card_dict = {}
card_list = map(int, input[3].split())

for card in map(int, input[1].split()):
    if card in card_dict:
        card_dict[card] += 1
    else:
        card_dict[card] = 1

result = [card_dict[card] if card in card_dict else 0 for card in card_list]

print(' '.join(map(str, result)))